test/rules/base.jl

```diff
@@ -0,0 +1,67 @@
+function test_scalar(f, f′, xs...)
+    for r = (rrule, frule)
+        rr = r(f, xs...)
+        @test rr !== nothing
+        fx, ∂x = rr
+        @test fx == f(xs...)
+        @test ∂x(0, 1) ≈ f′(xs...) atol=1e-5
+    end
+end
+
+@testset "Trig" begin
+    @testset "Basics" for x = (Float64(π), Complex(π, π/2))
+        test_scalar(sin, cos, x)
+        test_scalar(cos, x->-sin(x), x)
```

Comment:
`->-` is really annoying to read, my brain wants to interpret it as a line with arrow in the middle

```suggestion
        test_scalar(cos, x -> -sin(x), x)
```